### PR TITLE
Added a helper method to alias an expression

### PR DESCRIPTION
**Pull request by nolamesa**

### CODE REVIEW

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -2012,3 +2014,37 @@ def not_(expression, dialect=None, **opts):
...
+        alias (str or Identifier): the alias name to use. If the name has
+            special charachters it is quoted.
+        dialect (str): the dialect used to parse the input expression.
+        **opts: other options to use to parse the input expressions.
+
+    Returns:
+        Alias: the aliased expression
+    """
+    exp = _maybe_parse(expression, dialect=dialect, parser_opts=opts)
+    if isinstance(alias, Identifier):
```
Comment (barakalon) [inline sqlglot/expressions.py:2041]: `SAFE_IDENTIFIER_RE` feels like parser logic.

I think you can replace these few lines with:

```python
identifier = _maybe_parse(alias, parse_into=Identifier, dialect=dialect, parser_opts=opts)
```

We might also want to add some kind of check in `_maybe_parse`:
```
if isinstance(code_or_expression, Expression):
    if parse_into and not isinstance(code_or_expression, parse_into):
        raise ValueError(...)
    return code_or_expression
```
And add `Identifier` to the map here: https://github.com/tobymao/sqlglot/blob/main/sqlglot/parser.py#L261

Comment (barakalon) [reply sqlglot/expressions.py:2041]: Er, I think you can alias things that aren't identifiers?

e.g.
```SUM(foo) AS bar``

What are we trying to accomplish here? I'm assuming:

```python
select(alias("foo", "bar")).from("baz")
```
?

File: sqlglot/expressions.py [hunk 2]
```diff
@@ -2012,3 +2014,37 @@ def not_(expression, dialect=None, **opts):
...
+        identifier = alias
+    elif isinstance(alias, str):
+        quoted = not re.match(SAFE_IDENTIFIER_RE, alias)
+        identifier = Identifier(this=alias, quoted=quoted)
```
Comment (tobymao) [inline sqlglot/expressions.py:2045]: are you tied to doing safe quotes? pypika doesn't do this, it always quotes

File: sqlglot/expressions.py (hunk 1 above)
Comment (tobymao) [reply sqlglot/expressions.py:2041]: you can't parse_into an identifier for example

parse_one("x-1") you want this to be an identifier "x-1", but sqlglot doesn't know you intended it to be a string

File: sqlglot/expressions.py [hunk 3]
```diff
@@ -2012,3 +2014,37 @@ def not_(expression, dialect=None, **opts):
...
+    else:
+        raise ValueError(
```
Comment (tobymao) [inline sqlglot/expressions.py:2047]: yea maybe we don't need this, since it can be Sum(x) as y?

File: sqlglot/__init__.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
 from sqlglot.dialects import Dialect
 from sqlglot.errors import ErrorLevel, UnsupportedError, ParseError, TokenError
-from sqlglot.expressions import Expression, Select, and_, or_, not_, condition
+from sqlglot.expressions import Expression, Select, and_, or_, not_, condition, alias
```
Comment (tobymao) [inline sqlglot/__init__.py:3]: alias_ as alias

File: sqlglot/expressions.py (hunk 2 above)
Comment (nolamesa) [reply sqlglot/expressions.py:2045]: it felt a little cleaner than adding quotes everywhere, either way works for me

File: sqlglot/expressions.py [hunk 5]
```diff
@@ -2012,3 +2014,37 @@ def not_(expression, dialect=None, **opts):
         **opts,
     )
     return Not(this=_wrap_operator(this))
+
+
+SAFE_IDENTIFIER_RE = re.compile(r"^[a-zA-Z][\w]*$")
+
+
+def alias(expression, alias, dialect=None, **opts):
```
[the remark this answers is not in the document]
Comment (nolamesa) [reply sqlglot/expressions.py:2022]: hahah sure that can work fine

File: sqlglot/expressions.py (hunk 3 above)
Comment (nolamesa) [reply sqlglot/expressions.py:2047]: that would not make a difference, alias should still be an identifer or a string

Comment (tobymao) [reply sqlglot/expressions.py:2047]: ah you're right sorry

**Review by tobymao — APPROVED** [review]

**Pull request merged**